base_custom_attribute/models/attribute_attribute.py

```diff
@@ -36,18 +36,44 @@ class AttributeAttribute(models.Model):
     _inherits = {"ir.model.fields": "field_id"}
     _order = "sequence_group,sequence,name"
 
-    def _get_attrs(self):
+    def _remove_native_fields(self, eview):
+        """Remove native fields linked to attributes from eview and return them.
+        Be aware that it don't remove label or div elements related to these fields.
+        """
+        native_attrs = self.filtered(lambda att: not att.is_custom)
+        efields = []
+        for attr in native_attrs:
+            efield = eview.xpath("//field[@name='{}']".format(attr.name))[0]
+            efield.getparent().remove(efield)
+            efields.append(efield)
+        return efields
+
+    def _set_attrs(self):
         return {
             "invisible": [("attribute_set_id", "not in", self.attribute_set_ids.ids)]
         }
 
     @api.model
-    def _build_attribute_field(self, subgroup):
-        """Return an etree 'field' element made of the current attribute 'self'
-        and child of his etree group 'subgroup'"""
+    def _build_attribute_field(self, subgroup, native_efields):
+        """Add an etree 'field' subelement (related to the current attribute 'self')
+        to subgroup, with a conditional invisibility based on its attribute sets."""
         self.ensure_one()
         kwargs = {"name": "%s" % self.name}
-        kwargs["attrs"] = str(self._get_attrs())
+        kwargs["attrs"] = str(self._set_attrs())
+        kwargs["required"] = str(self.required or self.required_on_views)
+
+        if not self.is_custom:
+            if self._context.get("include_native_attribute"):
+                for native_efield in native_efields:
```

Comment:
I do not think this is necessary, This is too complex !
_build_attribute_field should do only one thing rendering an attribute.
Remove this native_efield stuff, we do not need it.